File: invesdwin-context-python-parent/invesdwin-context-python-runtime-contract/src/test/java/de/invesdwin/context/python/runtime/contract/InputsAndResultsTestFloat.py
print("getFloat")
if 'getFloat' in locals():
	raise Exception("getFloat already defined!")
getFloat = putFloat
getFloatType = type(getFloat)
print(getFloatType)
print(getFloat)
if(getFloatType is not float):
	raise Exception("getFloat not float!")

print("getFloatVector")
if 'getFloatVector' in locals():
	raise Exception("getFloatVector already defined!")
getFloatVector = putFloatVector
getFloatVectorType = type(getFloatVector[0])
print(getFloatVectorType)
print(getFloatVector)
if(getFloatVectorType is not float):
	raise Exception("getFloatVector not float!")

print("getFloatVectorAsList")
if 'getFloatVectorAsList' in locals():
	raise Exception("getFloatVectorAsList already defined!")
getFloatVectorAsList = putFloatVectorAsList
getFloatVectorAsListType = type(getFloatVectorAsList[0])
print(getFloatVectorAsListType)
print(getFloatVectorAsList)
if(getFloatVectorAsListType is not float):
	raise Exception("getFloatVectorAsList not float!")

print("getFloatMatrix")
if 'getFloatMatrix' in locals():
	raise Exception("getFloatMatrix already defined!")
getFloatMatrix = putFloatMatrix
getFloatMatrixType = type(getFloatMatrix[0][0])
print(getFloatMatrixType)
print(getFloatMatrix)
if(getFloatMatrixType is not float):
	raise Exception("getFloatMatrix not float!")

print("getFloatMatrixAsList")
if 'getFloatMatrixAsList' in locals():
	raise Exception("getFloatMatrixAsList already defined!")
getFloatMatrixAsList = putFloatMatrixAsList
getFloatMatrixAsListType = type(getFloatMatrixAsList[0][0])
print(getFloatMatrixAsListType)
print(getFloatMatrixAsList)
if(getFloatMatrixAsListType is not float):
	raise Exception("getFloatMatrixAsList not float!")

File: invesdwin-context-python-parent/invesdwin-context-python-runtime-contract/src/test/java/de/invesdwin/context/python/runtime/contract/InputsAndResultsTestFloat.java
package de.invesdwin.context.python.runtime.contract;

import java.util.ArrayList;
import java.util.List;

import javax.annotation.concurrent.NotThreadSafe;

import org.springframework.core.io.ClassPathResource;

import de.invesdwin.context.integration.script.IScriptTaskEngine;
import de.invesdwin.context.integration.script.IScriptTaskInputs;
import de.invesdwin.context.integration.script.IScriptTaskResults;
import de.invesdwin.util.assertions.Assertions;
import de.invesdwin.util.collections.Arrays;

@NotThreadSafe
public class InputsAndResultsTestFloat {

    private final IScriptTaskRunnerPython runner;

    public InputsAndResultsTestFloat(final IScriptTaskRunnerPython runner) {
        this.runner = runner;
    }

    public void testFloat() {
        //putFloat
        final float putFloat = 123.123F;

        //putFloatVector
        final float[] putFloatVector = new float[3];
        for (int i = 0; i < putFloatVector.length; i++) {
            putFloatVector[i] = Float.parseFloat((i + 1) + "." + (i + 1));
        }

        //putFloatVectorAsList
        final List<Float> putFloatVectorAsList = Arrays.asList(Arrays.toObject(putFloatVector));

        //putFloatMatrix
        final float[][] putFloatMatrix = new float[4][];
        for (int row = 0; row < putFloatMatrix.length; row++) {
            final float[] vector = new float[3];
            for (int col = 0; col < vector.length; col++) {
                vector[col] = Float.parseFloat((row + 1) + "." + (col + 1));
            }
            putFloatMatrix[row] = vector;
        }

        //putFloatMatrixAsList
        final List<List<Float>> putFloatMatrixAsList = new ArrayList<List<Float>>(putFloatMatrix.length);
        for (final float[] vector : putFloatMatrix) {
            putFloatMatrixAsList.add(Arrays.asList(Arrays.toObject(vector)));
        }

        new AScriptTaskPython<Void>() {

            @Override
            public void populateInputs(final IScriptTaskInputs inputs) {
                inputs.putFloat("putFloat", putFloat);

                inputs.putFloatVector("putFloatVector", putFloatVector);

                inputs.putFloatVectorAsList("putFloatVectorAsList", putFloatVectorAsList);

                inputs.putFloatMatrix("putFloatMatrix", putFloatMatrix);

                inputs.putFloatMatrixAsList("putFloatMatrixAsList", putFloatMatrixAsList);
            }

            @Override
            public void executeScript(final IScriptTaskEngine engine) {
                engine.eval(new ClassPathResource(InputsAndResultsTestFloat.class.getSimpleName() + ".py",
                        InputsAndResultsTestFloat.class));
            }

            @Override
            public Void extractResults(final IScriptTaskResults results) {
                //getFloat
                final Float getFloat = results.getFloat("getFloat");
                Assertions.assertThat(putFloat).isEqualTo(getFloat);

                //getFloatVector
                final float[] getFloatVector = results.getFloatVector("getFloatVector");
                Assertions.assertThat(putFloatVector).isEqualTo(getFloatVector);

                //getFloatVectorAsList
                final List<Float> getFloatVectorAsList = results.getFloatVectorAsList("getFloatVectorAsList");
                Assertions.assertThat(putFloatVectorAsList).isEqualTo(getFloatVectorAsList);

                //getFloatMatrix
                final float[][] getFloatMatrix = results.getFloatMatrix("getFloatMatrix");
                Assertions.assertThat(putFloatMatrix).isEqualTo(getFloatMatrix);

                //getFloatMatrixAsList
                final List<List<Float>> getFloatMatrixAsList = results.getFloatMatrixAsList("getFloatMatrixAsList");
                Assertions.assertThat(putFloatMatrixAsList).isEqualTo(getFloatMatrixAsList);
                return null;
            }
        }.run(runner);
    }

}
